### PR TITLE
BUG-104930 Periscope: input & runtime validation; cloudbreak: input v…

### PR DESCRIPTION
**Pull request by gergopapi2**
…alidation

BUG-104930 periscope: new env param renamed

BUG-104930 Fixing style issues; deleted unused test

BUG-104930 renaming ScalingHardLimits variables

BUG--104930 moved ValidationResult back to core

BUG-104930 Validation message refactored for scalingPolicy and scalingConfiguration

BUG-104930 fixing visibility for constant

### CODE REVIEW

File: autoscale-api/src/main/java/com/sequenceiq/periscope/api/endpoint/validator/ScalingPolicyRequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.sequenceiq.cloudbreak.common.type.ScalingHardLimits;
+import com.sequenceiq.cloudbreak.validation.ValidatorUtil;
+import com.sequenceiq.periscope.api.model.AdjustmentType;
+import com.sequenceiq.periscope.api.model.ScalingPolicyBase;
+
+/**
```
Comment (keyki) [inline autoscale-api/src/main/java/com/sequenceiq/periscope/api/endpoint/validator/ScalingPolicyRequestValidator.java]: Please remove this template from idea

File: core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java [hunk 2]
```diff
@@ -78,6 +79,9 @@
     @Inject
     private StackCreatorService stackCreatorService;
 
+    @Autowired
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java]: We use @Inject, please change the annotation

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.sequenceiq.cloudbreak.common.type;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by gpapp on 6/12/18.
+ */
+@Component
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java]: This should be a service

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public class ScalingHardLimits {
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java]: Name shold be xyService

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    private static final int DEFAULT_UPSCALING_MAXIMUM_NODE_COUNT = 100;
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java]: Static property not necessary

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java [hunk 6]
```diff
@@ -67,6 +72,11 @@ public void run() {
     }
 
     private void scaleUp(int scalingAdjustment, int totalNodes) {
+        if (scalingHardLimits.getMaxUpscaleStepInNodeCount() < scalingAdjustment) {
```
Comment (mhmxs) [inline autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java]: Code duplication 3 times

File: core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java (hunk 2 above)
Comment (gergopapi2) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java]: Ok, I will do that.

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java (hunk 5 above)
Comment (gergopapi2) [reply cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/type/ScalingHardLimits.java]: True, but then stylecheck complained: it does not like variables with underscore.

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java (hunk 6 above)
Comment (gergopapi2) [reply autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java]: Thanks!

File: autoscale-api/src/main/java/com/sequenceiq/periscope/api/endpoint/validator/ScalingPolicyRequestValidator.java (hunk 1 above)
Comment (gergopapi2) [reply autoscale-api/src/main/java/com/sequenceiq/periscope/api/endpoint/validator/ScalingPolicyRequestValidator.java]: Done

**Pull request merged**